org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/typesystem/computation/XbaseTypeComputer.java

```diff
@@ -1166,17 +1166,18 @@ protected void _computeTypes(XTryCatchFinallyExpression object, ITypeComputation
 				caughtExceptions.add(referenceOwner.toLightweightTypeReference(catchClause.getDeclaredParam().getParameterType()));
 		}
 		
+		ITypeComputationState withExpectedExceptions = state.withExpectedExceptions(caughtExceptions);
```

Comment:
do i see it correct we dont have a negative test here?

Reply:
I’ll add a linking test